### PR TITLE
chore: refactor jsobject response & settings

### PR DESCRIPTION
**Pull request by ashit-rath**
## Description
Refactor the JSObject response view and the settings to accept new columns

#### PR fixes following issue(s)
Fixes for PR https://github.com/appsmithorg/appsmith-ee/pull/3060
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
  - Users can now opt to open the debugger when executing JavaScript functions within the app.

- **Enhancements**
  - Improved JavaScript function execution flow with additional debugging options.
  - Enhanced JavaScript editor form to better handle JavaScript collection configurations.

- **Bug Fixes**
  - Fixed issues related to JavaScript collection data retrieval and management.

- **Refactor**
  - Streamlined JavaScript function settings with additional columns and headings for better clarity.
  - Updated state management to align with the new JavaScript collection data structure.

- **Style**
  - Removed the bottom border from the `TabbedViewContainer` for a cleaner UI appearance.

- **Documentation**
  - Updated documentation to reflect changes in JavaScript pane actions and selectors.

### CODE REVIEW

File: app/client/src/sagas/JSPaneSagas.ts [hunk 1]
```diff
@@ -484,9 +485,10 @@ export function* handleStartExecuteJSFunctionSaga(
     action: JSAction;
     collection: JSCollection;
     from: EventLocation;
+    openDebugger?: boolean;
   }>,
-) {
-  const { action, collection, from } = data.payload;
+): any {
```
Comment (ankitakinger) [inline app/client/src/sagas/JSPaneSagas.ts:490]: The type `any` is intentional here? Should we have a defined type instead?

File: app/client/src/selectors/editorSelectors.tsx [hunk 2]
```diff
@@ -920,6 +920,21 @@ export const getActionById = createSelector(
   },
 );
 
+export const getJSCollectionDataById = createSelector(
+  [
+    getJSCollections,
+    (state: any, props: any) => props.match.params.collectionId,
```
Comment (ankitakinger) [inline app/client/src/selectors/editorSelectors.tsx:926]: Types here as well

File: app/client/src/sagas/JSPaneSagas.ts (hunk 1 above)
Comment (ashit-rath) [reply app/client/src/sagas/JSPaneSagas.ts:490]: Not intentional; fixing

File: app/client/src/selectors/editorSelectors.tsx (hunk 2 above)
Comment (ashit-rath) [reply app/client/src/selectors/editorSelectors.tsx:926]: It's a copied code but let me fix.

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**